nntrainer/models/neuralnet.cpp

```diff
@@ -203,10 +203,6 @@ int NeuralNetwork::initialize() {
 
   initialized = true;
 
-  if (!load_path.empty()) {
```

Comment:
this will break the inference API. I would recommend loading the weights here and then loading again at training for the optimizer.

To handle the optimizer variables not being allocated, you can move the file pointer ahead by the size than reading those values.

Reply:
I think it is time to consider generating the model and bin file for the inference and training separately. For the temporary fixes, we could move ahead as you said for the case of inference. On second thought, need to handle different optimizers. for example, the model is saved with ADAM but load with SGD. We need to write the opt type in the model file. Definitely need some kind of header.

Reply:
IMO the inference and training saved file should be compatible for both training and inference.
Overall, we need a structure to the model file - flatbuffer/json format or some header.

Reply:
right. compatible but no need to be the same file.